File: Programming Language Performance Tests/Python/DatabasePerformanceTest.py
from sqlalchemy import create_engine


class UnicodeEntry:
    def __init__(self, ID: int, UnicodeCharacter: str, PREVIOUS: int, NEXT: int):
        self.ID = ID
        self.UnicodeCharacter = UnicodeCharacter
        self.PREVIOUS = PREVIOUS
        self.NEXT = NEXT


def execute():
    unicode_entry_list = []

    try:
        engine = create_engine("sqlite:///unicode.sqlite")
        connection = engine.connect()
        unicode_characters = connection.execute("SELECT * FROM unicode;")
        for row in unicode_characters:
            unicode_entry_list.append(
                UnicodeEntry(
                    row[0],
                    row[1],
                    row[2],
                    row[3]
                )
            )
    except Exception as e:
        print(e)

File: Programming Language Performance Tests/CSharp/Database_Performance_Test.cs
using System;
using System.Collections.Generic;
using System.Data.SQLite;

namespace CSharpPerformanceTest
{
    public class DatabasePerformanceTest
    {
        class UnicodeEntry
        {
            public long ID { get; set; }
            public char UnicodeCharacter { get; set; }
            public long PREVIOUS { get; set; }
            public long NEXT { get; set; }

            public UnicodeEntry(long ID, char UnicodeCharacter, long PREVIOUS, long NEXT)
            {
                this.ID = ID;
                this.UnicodeCharacter = UnicodeCharacter;
                this.PREVIOUS = PREVIOUS;
                this.NEXT = NEXT;
            }
        }

        public void Execute()
        {
            var unicodeEntryList = new List<UnicodeEntry>();
            try
            {
                SQLiteConnection connection = new SQLiteConnection("Data Source=unicode.sqlite;");
                connection.Open();
                var command = new SQLiteCommand("SELECT * FROM unicode;", connection);
                SQLiteDataReader reader = command.ExecuteReader();
                while (reader.Read())
                {
                    string resultString = reader.GetString(1);
                    unicodeEntryList.Add(new UnicodeEntry(
                        reader.IsDBNull(0) ? 0 : reader.GetInt64(0),
                        resultString.Length > 0 ? resultString[0] : ' ',
                        reader.IsDBNull(2) ? 0 : reader.GetInt64(2),
                        reader.IsDBNull(3) ? 0 : reader.GetInt64(3)
                    ));
                }
            }
            catch (SQLiteException e)
            {
                Console.WriteLine(e);
            }
        }
    }
}